db/types.c

```diff
@@ -13400,11 +13400,23 @@ int real_to_dttz(double d, dttz_t *dt, int precision)
     dt->dttz_conv = 1;
     dt->dttz_sec = d;
     if ((long long)(d * 1000) == d * 1000) {
-        dt->dttz_frac = (d - (long long)d) * 1E3 + 0.5;
+        dt->dttz_frac = (d - (long long)d) * 1E3;
```

Comment:
since we already casted for the assignment in 13401, maybe change to:
  dt->dttz_frac = (d - dt->dttz_sec) * 1E3;